rc/tools/go/gopls.kak

```diff
@@ -0,0 +1,49 @@
+hook -once global BufSetOption filetype=go %{
+    require-module gopls
+}
+
+evaluate-commands %sh{
+    if ! command -v gopls > /dev/null 2>&1; then
+        echo "fail Please install gopls or add to PATH!"
+    fi
+}
+
+define-command -docstring "Enter gopls mode.
+Bindings for gopls in kakoune.
+
+Best used with mapping like:
+	map global normal '<whatever>' ': gopls-mode<ret>'
+" \
+gopls-mode %{ require-module gopls; evaluate-commands 'enter-user-mode gopls' }
+
+provide-module gopls %§
+
+try %{ declare-user-mode gopls }
+
+# Auto-format and auto-imports
+# ----------------------------
+
+map global gopls -docstring "format file" 'f' '<esc>: gopls-cmd format<ret>'
+map global gopls -docstring "automatic imports" 'i' '<esc>: gopls-cmd imports<ret>'
+
+declare-option -hidden str gopls_tmp_dir
+
+define-command -hidden -params 1 gopls-cmd \
+    -docstring "run various gopls commands" %{
+    evaluate-commands %sh{
+        dir=$(mktemp -d "${TMPDIR:-/tmp}"/kak-gopls.XXXXXXXX)
+        printf %s\\n "set-option buffer gopls_tmp_dir ${dir}"
+    }
+    evaluate-commands %sh{
+        dir=${kak_opt_gopls_tmp_dir}
+        eval "gopls $1 -w ${kak_buffile} 2> ${dir}/stderr"
+        if [ ! $? ]; then
```

Comment:
This condition is always false, use `[ $? -ne 0 ]`

Reply:
Fixed, thanks.